lib/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart

```diff
@@ -31,9 +38,19 @@ class MailboxDashBoardController extends BaseController {
 
   void _setSessionCurrent() {
     sessionCurrent = Get.arguments as Session;
-    if (Get.isRegistered<MailboxController>()) {
-      final mailboxController = Get.find<MailboxController>();
-      mailboxController.setAccountId(sessionCurrent.accounts.keys.first);
+    accountId = sessionCurrent?.accounts.keys.first;
+    if (accountId != null) {
+      mailboxController.getAllMailboxAction(accountId!);
     }
   }
+
+  void setSelectedMailbox(PresentationMailbox newPresentationMailbox) {
+    selectedMailbox.value = newPresentationMailbox;
+
+    threadController.refreshGetAllEmailAction();
```

Comment:
please check the comment: https://github.com/linagora/tmail-flutter/pull/21#discussion_r690105219